jgrapht-core/src/test/java/org/jgrapht/alg/matching/ApproximateWeightedMatchingTest.java

```diff
@@ -0,0 +1,349 @@
+        Graphs.addEdge(g, 2, 2, -200.0);
+        Graphs.addEdge(g, 3, 3, -100.0);
+        Graphs.addEdge(g, 4, 4, 0.0);
+
+        WeightedMatchingAlgorithm<Integer, DefaultWeightedEdge> mm = getApproximationAlgorithm(g);
+
+        assertEquals(4, mm.getMatching().size());
+        assertEquals(4.0, mm.getMatchingWeight(), PathGrowingWeightedMatching.DEFAULT_EPSILON);
+        assertTrue(isMatching(g, mm.getMatching()));
+    }
+
+    public void testMultiGraph()
+    {
+        WeightedPseudograph<Integer, DefaultWeightedEdge> g =
+            new WeightedPseudograph<>(DefaultWeightedEdge.class);
+
+        Graphs.addAllVertices(g, Arrays.asList(0, 1, 2, 3));
+        Graphs.addEdge(g, 0, 1, 1.0);
+        Graphs.addEdge(g, 1, 2, 1.0);
+        Graphs.addEdge(g, 2, 3, 1.0);
+        Graphs.addEdge(g, 3, 0, 1.0);
+        Graphs.addAllVertices(g, Arrays.asList(4, 5, 6, 7));
+        Graphs.addEdge(g, 4, 5, 1.0);
+        Graphs.addEdge(g, 5, 6, 1.0);
+        Graphs.addEdge(g, 6, 7, 1.0);
+        Graphs.addEdge(g, 7, 4, 1.0);
+
+        // add multiple edges
+        Graphs.addEdge(g, 0, 1, 2.0);
+        Graphs.addEdge(g, 1, 2, 2.0);
+        Graphs.addEdge(g, 2, 3, 2.0);
+        Graphs.addEdge(g, 3, 0, 2.0);
+        Graphs.addEdge(g, 4, 5, 2.0);
+        Graphs.addEdge(g, 5, 6, 2.0);
+        Graphs.addEdge(g, 6, 7, 2.0);
+        Graphs.addEdge(g, 7, 4, 2.0);
+
+        WeightedMatchingAlgorithm<Integer, DefaultWeightedEdge> mm = getApproximationAlgorithm(g);
+
+        // greedy finds maximum here 8.0
+        assertEquals(4, mm.getMatching().size());
+        assertEquals(8.0, mm.getMatchingWeight(), PathGrowingWeightedMatching.DEFAULT_EPSILON);
+        assertTrue(isMatching(g, mm.getMatching()));
+    }
+
+    public void testDirected()
+    {
+        DirectedWeightedPseudograph<Integer, DefaultWeightedEdge> g =
+            new DirectedWeightedPseudograph<>(DefaultWeightedEdge.class);
+
+        Graphs.addAllVertices(g, Arrays.asList(0, 1, 2, 3));
+        Graphs.addEdge(g, 0, 1, 1.0);
+        Graphs.addEdge(g, 1, 2, 1.0);
+        Graphs.addEdge(g, 2, 3, 1.0);
+        Graphs.addEdge(g, 3, 0, 1.0);
+        Graphs.addAllVertices(g, Arrays.asList(4, 5, 6, 7));
+        Graphs.addEdge(g, 4, 5, 1.0);
+        Graphs.addEdge(g, 5, 6, 1.0);
+        Graphs.addEdge(g, 6, 7, 1.0);
+        Graphs.addEdge(g, 7, 4, 1.0);
+
+        // add multiple edges
+        Graphs.addEdge(g, 0, 1, 2.0);
+        Graphs.addEdge(g, 1, 2, 2.0);
+        Graphs.addEdge(g, 2, 3, 2.0);
+        Graphs.addEdge(g, 3, 0, 2.0);
+        Graphs.addEdge(g, 4, 5, 2.0);
+        Graphs.addEdge(g, 5, 6, 2.0);
+        Graphs.addEdge(g, 6, 7, 2.0);
+        Graphs.addEdge(g, 7, 4, 2.0);
+
+        WeightedMatchingAlgorithm<Integer, DefaultWeightedEdge> mm = getApproximationAlgorithm(g);
+
+        assertEquals(4, mm.getMatching().size());
+        assertEquals(8.0, mm.getMatchingWeight(), PathGrowingWeightedMatching.DEFAULT_EPSILON);
```

Comment:
ok, maybe replace the occurrences of PathGrowingWeightedMatching.DEFAULT_EPSILON by WeightedMatchingAlgorithm .DEFAULT_EPSILON